hyperspy/axes.py

```diff
@@ -504,6 +504,20 @@ def __getitem__(self, y):
         """x.__getitem__(y) <==> x[y]
 
         """
+        return_one = False
+        if isinstance(y, basestring) or not np.iterable(y):
+            y = (y,)
+            return_one = True
```

Comment:
Maybe more elegant to replace `return_one = True` with `return self[(y,)][0]` ? Simplifies the rest of the function as well, but relies on recursion (might be an issue with inheritance).